lib/internal/bootstrap/node.js

```diff
@@ -314,6 +314,14 @@ if (getOptionValue('--experimental-report')) {
   }
 }
 
+// This is a shared domains-like stack that is used by any domain-like facility
+// to push domain-like instances on a common stack. Sharing this among all
+// domain-like implementations means that those implementations can be used
+// concurrently in the same process space, and e.g. one domain-like
+// implementation can restore another domain-like implementation's instance when
+// exiting.
+process._domainsStack = [];
```

Comment:
if this is meant to be private please don't expose it on process

Reply:
Right, this is more a basic implementation for now that is not encapsulated to propose the general concept of a shared stack. I agree this should be encapsulated somehow.

Reply:
@devsnek Looking at this once again, just to make sure we're on the same page: this stack is meant to be shared across implementations of domain-like modules loaded within the same process.

I set it on the `process` object for now because it was quick and good enough to convey this intention of having a shared stack, but ideally it could be scoped to a different object/subsystem that is more specific.

I'm not sure what would be the best way to do that though, so I'm definitely open to suggestions. A few other ideas I had in mind:

1. keeping it in the core `domain`, but making pushing to/popping from it possible via public module functions. This is not necessarily ideal because:
    1. it means that loading the core domain module shouldn't enable some of its custom behavior, which is what currently happens. So some refactoring would be needed to make this work, and the core `domain` API might need to change to have some sort of `enable()` API.
    1. it means that any domain-like implementation would depend on the core `domain` module, which I'm not sure is what we want.

1. moving it to the `Environment`. I'm not sure of all the implications of that, but it seems that would require user-land modules to use the native add-on API, which is not ideal.

Any thoughts on that?

Reply:
to clarify, i have no opinion on the specifics of this implementation. I just bring this up because we no longer use underscored properties in core. if this property is meant to be private it should be encapsulated such that userland cannot touch it. if it is meant to be public it should not have an underscore. whether it is on process or domain or whatever is beyond what i am asking.